04.bowling/bowling.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+score = ARGV[0]
+scores = score.to_s.split(',')
+shots = []
+scores.each do |s|
+  if s == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << s.to_i
+  end
+end
+
+frames = []
+shots.each_slice(2) do |s|
+  frames << s
+end
+indexes = [*(1..12)]
+i_f_hash = Hash[indexes.zip frames]
+
+n = 0
+point = 0
+i_f_hash.each do
+  if i_f_hash[n + 1][0] == 10
```

Comment:
ここから ifの分岐が多くて読むのが難しいので、適度にコメントを入れていきましょう。

```suggestion
  if i_f_hash[n + 1][0] == 10 # ストライクの時
```